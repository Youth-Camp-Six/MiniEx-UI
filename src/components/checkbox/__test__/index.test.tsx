// /**
//  * @jest-environment jsdom
//  */

// import React from 'react';
// import { render } from '@testing-library/react';
// import '@testing-library/jest-dom/extend-expect';
// import { CheckboxGroup } from '..';

// const checkBoxDataList = [
//   { label: 'labe-0', value: 0 },
//   { label: 'labe-1', value: 1 },
//   { label: 'labe-2', value: 2 },
// ];

// describe('test CheckboxGroup comonent', () => {
//   it('checkbox snapshot test', () => {
//     const onChange = jest.fn();

//     const wrapper = render(
//       <CheckboxGroup
//         value={[1, 2, 3]}
//         options={checkBoxDataList}
//         onChange={onChange}
//         round
//       ></CheckboxGroup>
//     );

//     expect(wrapper.asFragment()).toMatchSnapshot();
//   });

//   it('renders checkbox options and updates selected value', () => {
//     const onChange = jest.fn();
//     const wrapper = render(
//       <CheckboxGroup
//         value={[0, 2]}
//         options={checkBoxDataList}
//         onChange={onChange}
//         round
//       ></CheckboxGroup>
//     );

//     const labe0 = wrapper.getByText('labe-0');
//     const labe1 = wrapper.getByText('labe-1');
//     const labe2 = wrapper.getByText('labe-2');

//     expect(labe0).toHaveClass('mi-checkbox-checked');
//     expect(labe1).not.toHaveClass('mi-checkbox-checked');
//     expect(labe2).not.toHaveClass('mi-radio-checked');
//     expect(onChange).not.toHaveBeenCalled();

//     option0.click();
//     expect(option0).toHaveClass('mi-radio-checked');
//     expect(option1).not.toHaveClass('mi-radio-checked');
//     expect(option2).not.toHaveClass('mi-radio-checked');
//     expect(onChange).toHaveBeenCalledWith('0');
//   });
// });
