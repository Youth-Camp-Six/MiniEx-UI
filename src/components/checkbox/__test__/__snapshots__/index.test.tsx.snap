// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CheckboxGroup comonent CheckboxGroup snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mi-flex mi-checkbox mi-checkbox-group-x mi-checkbox-group-medium mi-flex-x mi-flex-x-start mi-flex-y-center"
    style="gap: 15px;"
  >
    <label
      class="mi-checkbox 
        mi--checkbox-block
        mi-checkbox-round
        mi-checkbox-checked
        mi-checkbox-medium
        
        "
    >
      <div
        class="icon-checkbox-box"
      >
        <svg
          class="icon-checkbox"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="4"
          viewBox="0 0 24 24"
          width="24"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <input
        checked=""
        style="display: none;"
        type="checkbox"
      />
      <span
        class="mi-checkbox-label"
      >
        labe1
      </span>
    </label>
    <label
      class="mi-checkbox 
        mi--checkbox-block
        mi-checkbox-round
        mi-checkbox-checked
        mi-checkbox-medium
        
        "
    >
      <div
        class="icon-checkbox-box"
      >
        <svg
          class="icon-checkbox"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="4"
          viewBox="0 0 24 24"
          width="24"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <input
        checked=""
        style="display: none;"
        type="checkbox"
      />
      <span
        class="mi-checkbox-label"
      >
        labe2
      </span>
    </label>
    <label
      class="mi-checkbox 
        mi--checkbox-block
        mi-checkbox-round
        mi-checkbox-checked
        mi-checkbox-medium
        
        "
    >
      <div
        class="icon-checkbox-box"
      >
        <svg
          class="icon-checkbox"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="4"
          viewBox="0 0 24 24"
          width="24"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <input
        checked=""
        style="display: none;"
        type="checkbox"
      />
      <span
        class="mi-checkbox-label"
      >
        labe3
      </span>
    </label>
    <label
      class="mi-checkbox 
        mi--checkbox-block
        mi-checkbox-round
        
        mi-checkbox-medium
        
        "
    >
      <div
        class="icon-checkbox-box"
      >
        <svg
          class="icon-checkbox"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="4"
          viewBox="0 0 24 24"
          width="24"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <input
        style="display: none;"
        type="checkbox"
      />
      <span
        class="mi-checkbox-label"
      >
        labe4
      </span>
    </label>
  </div>
</DocumentFragment>
`;

exports[`test CheckboxGroup comonent checkbox snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mi-flex mi-checkbox mi-checkbox-group-x mi-checkbox-group-medium mi-flex-x mi-flex-x-start mi-flex-y-center"
    style="gap: 15px;"
  >
    <label
      class="mi-checkbox 
        
        mi-checkbox-round
        mi-checkbox-checked
        mi-checkbox-medium
        
        "
    >
      <div
        class="icon-checkbox-box"
      >
        <svg
          class="icon-checkbox"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="4"
          viewBox="0 0 24 24"
          width="24"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <input
        checked=""
        style="display: none;"
        type="checkbox"
      />
      <span
        class="mi-checkbox-label"
      >
        labe1
      </span>
    </label>
    <label
      class="mi-checkbox 
        
        mi-checkbox-round
        mi-checkbox-checked
        mi-checkbox-medium
        
        "
    >
      <div
        class="icon-checkbox-box"
      >
        <svg
          class="icon-checkbox"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="4"
          viewBox="0 0 24 24"
          width="24"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <input
        checked=""
        style="display: none;"
        type="checkbox"
      />
      <span
        class="mi-checkbox-label"
      >
        labe2
      </span>
    </label>
    <label
      class="mi-checkbox 
        
        mi-checkbox-round
        mi-checkbox-checked
        mi-checkbox-medium
        
        "
    >
      <div
        class="icon-checkbox-box"
      >
        <svg
          class="icon-checkbox"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="4"
          viewBox="0 0 24 24"
          width="24"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <input
        checked=""
        style="display: none;"
        type="checkbox"
      />
      <span
        class="mi-checkbox-label"
      >
        labe3
      </span>
    </label>
    <label
      class="mi-checkbox 
        
        mi-checkbox-round
        
        mi-checkbox-medium
        
        "
    >
      <div
        class="icon-checkbox-box"
      >
        <svg
          class="icon-checkbox"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="4"
          viewBox="0 0 24 24"
          width="24"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <input
        style="display: none;"
        type="checkbox"
      />
      <span
        class="mi-checkbox-label"
      >
        labe4
      </span>
    </label>
  </div>
</DocumentFragment>
`;
